### PR TITLE
Did some refactoring and additions in Datasources code.

### PR DESCRIPTION
**Pull request by hhovsepy**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d3d71ab843232abaa10d421062cdecbbb4956013:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/provider/hawkular.py [hunk 1]
```diff
@@ -93,13 +93,13 @@ def num_datasource(self):
     @num_datasource.variant('ui')
     def num_datasource_ui(self, reload_data=True):
         if reload_data:
-            self.summary.reload()
+            self.load_details()
```
Comment (mfalesni) [inline cfme/middleware/provider/hawkular.py]: I guess you understood my recommendation wrong, summary reload is necessary if you want to refresh the values in summary, which you **want** to do in this method ...

**Comment by dajoRH** [comment]
Lint report for commit 45a5cd1b59887a0ff828feb46cdff0e19e492615:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/middleware/datasource.py [hunk 2]
```diff
@@ -41,12 +41,12 @@ def _db_select_query(name=None, nativeid=None, server=None, provider=None):
 
 def _get_datasources_page(provider=None, server=None):
     if server:  # if server instance is provided navigate through server page
-        server.summary.reload()
+        server.load_details()
```
Comment (mfalesni) [inline cfme/middleware/datasource.py]: this is also unnecessary change as you click in the summary item the next thing. So usag eofsummary reload is a good idea here.

**Comment by dajoRH** [comment]
Lint report for commit 21872def75a2502a90c64a2ad1f995f4e2865eae:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by hhovsepy** [comment]
The idea of my changes is to avoid unnecessary reloads, as load_details() checks whether  it is on summary page, it does not reload.

**Comment by dajoRH** [comment]
Lint report for commit 84bd59df51d9b8c77f3bf9446884175e85f0f273:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by mfalesni** [comment]
If you are using summary, then you must use summary reload. Otherwise it is not recommended since load_details is the thing that ensures you get to the summary page.

With the refresh, you can tell load_details to refresh or not ...

**Comment by dajoRH** [comment]
Lint report for commit 55a95d2df3f91032d5bad511bb6eb6c02fc8e9d9:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 81c91dd3586c25bc170f309feebeee8c2896cd74:

Everything seems all right :smile: .
*CFME QE Bot*
